musig/src/aggregated_pubkey.rs

```diff
@@ -0,0 +1,53 @@
+use crate::errors::MusigError;
+use crate::hasher::Hasher;
+use bellman::{Field, PrimeField};
+use franklin_crypto::eddsa::PublicKey;
+use franklin_crypto::jubjub::edwards::Point;
+use franklin_crypto::jubjub::{JubjubEngine, ToUniform};
+
+pub struct AggregatedPublicKey;
+
+impl AggregatedPublicKey {
+    pub(crate) fn compute_from_pubkeys<E: JubjubEngine>(
+        pubkeys: &[PublicKey<E>],
+        position: usize,
+        jubjub_params: &<E as JubjubEngine>::Params,
+    ) -> Result<(PublicKey<E>, Vec<E::Fs>), MusigError> {
+        if pubkeys.is_empty() {
+            return Err(MusigError::InvalidPubkeyLength);
+        }
+
+        if position >= pubkeys.len() {
+            return Err(MusigError::InvalidParticipantPosition);
+        }
+        for pubkey in pubkeys {
+            // check that pubkey is in correct subgroup
+            if pubkey.0.mul(E::Fs::char(), jubjub_params) != Point::zero() {
+                return Err(MusigError::InvalidPublicKey);
+            }
+        }
+
+        // TODO: sort pubkeys
```

Comment:
You should either implement it, or require on outer level that aggregated key may differ depending on the order of public keys in a set. Also, in deserialization you should check that encoding is unique (elements are in a field). I think it's done for your already, but check it just in case.